core/src/main/java/com/segment/analytics/kotlin/core/Telemetry.kt

```diff
@@ -0,0 +1,272 @@
+package com.segment.analytics.kotlin.core
+
+import kotlinx.serialization.*
+import kotlinx.serialization.json.Json
+import kotlinx.coroutines.*
+import java.net.HttpURLConnection
+import java.lang.System
+import java.time.LocalDateTime
+import kotlin.math.min
+import kotlin.math.roundToInt
+
+class MetricsRequestFactory : RequestFactory() {
+    override fun upload(apiHost: String): HttpURLConnection {
+        val connection: HttpURLConnection = openConnection("https://${apiHost}/m")
+        connection.setRequestProperty("Content-Type", "text/plain")
+        connection.doOutput = true
+        connection.setChunkedStreamingMode(0)
+        return connection
+    }
+}
+
+@Serializable
+data class RemoteMetric(
+    val type: String,
+    val metric: String,
+    var value: Int,
+    val tags: Map<String, String>,
+    val log: Map<String, String>? = null,
+)
+
+fun logError(err: Throwable) {
+    println("Error sending segment performance metrics $err")
+}
+
+object Telemetry {
+    private var _enable: Boolean = true
+    private var _host: String = Constants.DEFAULT_API_HOST
+    private var _sampleRate: Double = 1.0
+    private var _flushTimer: Int = 30 * 1000 // 30s
+    private var _maxQueueSize: Int = 20
+    private var MAX_QUEUE_BYTES = 28000
+    private var _maxQueueBytes: Int = MAX_QUEUE_BYTES
+    private var _httpClient: HTTPClient = HTTPClient("", MetricsRequestFactory())
+    private var _sendWriteKeyOnError: Boolean = true
+    private var _sendErrorLogData: Boolean = false
+    private var _errorHandler: ((Throwable) -> Unit)? = null
+    private var _rateLimitTimer: Int = 0
+
+    var enable: Boolean
+        get() = _enable
```

Comment:
no need to write getter/setter out, by default it's property, the compiler generates getter/setter and field

Reply:
Do you have to remove the "_" to indicate a backing property and make it non private?

I guess if the only thing we're doing is passing the value without modification it makes sense to make a regular property.